utils/objectUtils.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Checks if an object has a property with the given path.
+ * @param {Object} obj - The object to check.
+ * @param {string | Array<string>} path - The path to the property to check. Can be a string with dot notation or an array of keys.
+ * @returns {boolean} - True if the object has the property, false otherwise.
+ */
+const hasProperty = (obj, path) => {
+  /**
+   * This regexp is used to split the path string into an array of keys using "[", "]" and "." as separators.
+   * In https://regexr.com/58j0k you can get a playground to test and analyze it.
+   * @type {RegExp}
+   * @example
+   * // returns ["test", "test2", "test3"]
+   * "test[test2].test3".match(/([^[.\]])+/g)
+   */
+  const pathArray = Array.isArray(path) ? path : path.match(/([^[.\]])+/g)
```

Comment:
This sometimes returns a string, so maybe `path` should be converted to an array?

Also, take care of the case `.match` return null

![image](https://github.com/victor-heliomar/i18n-translation-generator/assets/36519478/28817ace-a06f-4db5-bb0b-af0d2b4b9670)

Reply:
Can you please explain a bit more about this point?: "This sometimes returns a string, so maybe path should be converted to an array?" I think it is converting it to array when receives an string, is that what you meant?

Reply:
Will check .match return null, didn't notice that. Thanks!

Reply:
Added a guard to avoid the program crash when match returns null. Thanks for notice it!